lib/Optimizer/CodeGen/QuakeToLLVM.cpp

```diff
@@ -1213,6 +1214,137 @@ class RemoveRelaxSizeRewrite : public OpConversionPattern<quake::RelaxSizeOp> {
     return success();
   }
 };
+
+/// NOTE: This class currently targets simulator backends. On hardware targets
+/// the custom operations ought to be decomposed by a separate pass and should
+/// never reach here.
+class CustomUnitaryOpRewrite
```

Comment:
We should probably add a FileCheck test for this in `test/Quake-QIR`

Reply:
Addressed in e137ebef99049d0cd363821a857064b683840069